### ISSUE TITLE
[$500] Public room - Focus mode pop-up opens if navigate to public room as anonymous user

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34740]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.27-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Precondition: user is logged in with account that has many reports, not a new account
1. Go to https://staging.new.expensify.com/
2. Go to incognito mode (close and reopen if it is open)
3. Log in
4. Create a public room
5. Send a reply in the room and also reply in a thread
6. Copy public room link
7. Log out
8. Navigate via copied public room link

## Expected Result:
Focus mode pop-up not opens if navigate to public room as anonymous user

## Actual Result:
Focus mode pop-up opens if navigate to public room as anonymous user

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/5ace35d2-7e9f-4e64-a20f-bfd85b9f1610

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01150febaacdca6d76</li>
        <li>Upwork Job ID: 1748014073803857920</li>
        <li>Last Price Increase: 2024-02-08</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
[#vip-vsb](https://expensify.enterprise.slack.com/archives/C066HJM2CAZ)

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Focus mode pop-up opens if navigate to public room as anonymous user
### What is the root cause of that problem?
Focus mode from backend  is returned as `true` [here](https://github.com/Expensify/App/blob/dcec4c2eddf70e5dc9491acc9f2e2a80c4ded5e7/src/Expensify.js#L273-L276) 
  and thus the focus mode is shown in [this line](https://github.com/Expensify/App/blob/dcec4c2eddf70e5dc9491acc9f2e2a80c4ded5e7/src/Expensify.js#L231) 
### What changes do you think we should make in order to solve the problem?
add another condition to validate that the user is authenticated

```js
                    {props.focusModeNotification && isAuthenticated ? <FocusModeNotification /> : null}
```

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Public room - Focus mode pop-up opens if navigate to public room as anonymous user
### What is the root cause of that problem?
The above proposal describes the root cause correct
### What changes do you think we should make in order to solve the problem?
I suggest to calculate the `reportCount` another way.  we must count those have a `reportId` key .
to do that we can change this [line](https://github.com/Expensify/App/blob/599c005849cd065a53d35e015991b637c768f2da/src/libs/actions/PriorityMode.ts#L123C9-L123C66) with below code
```
const reportCount =  _.filter(allReports ?? {}, obj => _.has(obj, 'reportId')).length
```

### What alternative solutions did you explore? (Optional)

**Comment by slafortune** [comment]
@eVoloshchak is there a proposal we can move forward with here?

**Comment by eVoloshchak** [comment]
I think we should proceed with @tienifr's [proposal](https://github.com/Expensify/App/issues/34740#issuecomment-1898876981)

While @abzokhattab's proposal is simpler and does prevent the issue from happening, it doesn't resolve the root cause (incorrect report count)

🎀👀🎀 C+ reviewed!

**Comment by shahinyan11** [comment]
@eVoloshchak I would also like to see feedback on my proposal

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by eVoloshchak** [comment]
Same, unable to reproduce the issue anymore
Let's wait for the second week KI tests then

**Comment by slafortune** [comment]
waiting for another test

**Comment by slafortune** [comment]
One more test this week

**Comment by marcaaron** [comment]
WRT to the issue. I am not too convinced we need to optimize this for anonymous users they are likely not going to have large numbers reports before becoming a real user.

>Onyx is retaining the reports data even after the user logged out

I agree with @tienifr that this continues to be a strange problem to diagnose and has been around for a while. If we can get a reproduction then it would be good to fix. I tried to raise an issue about this but it was closed so I'm not sure it's really a near term priority for us.

>security issue since Onyx is retaining the reports data even after the user logged out

Thats an interesting take. I agree it's a problem in theory as someone could use the app on a public computer or network and that behavior would be very concerning. However, we really need a reproduction. I think it requires a lot of data to exist in Onyx for this to happen. And haven't been able to reliably reproduce it.
